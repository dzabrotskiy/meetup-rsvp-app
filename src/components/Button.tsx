import { StyleSheet, TouchableOpacity } from 'react-native';

import { primaryColor } from 'constants/Colors';

import { Text } from './Themed';

type Props = {
  title?: string;
  onPress?(): void;
  type?: 'primary' | 'secondary' | 'tertiary';
};

export function Button({ title, onPress, type = 'primary' }: Props) {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.container, type === 'secondary' && styles.secondary]}
      activeOpacity={0.9}
    >
      <Text style={[styles.title, type === 'primary' && styles.titlePrimary]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 45,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: primaryColor,
    borderRadius: 11,
  },
  title: {
    fontSize: 17,
  },
  titlePrimary: {
    color: '#fff',
  },
  secondary: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: primaryColor,
  },
});
