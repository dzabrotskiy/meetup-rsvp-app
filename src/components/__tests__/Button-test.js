import * as React from 'react';
import renderer from 'react-test-renderer';
import { TouchableOpacity } from 'react-native';

import { Button } from '../Button';
import { Text } from '../Themed';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button title="Press Me" onPress={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('should call onPress', () => {
  const onPress = jest.fn();
  const tree = renderer.create(<Button title="Press Me" onPress={onPress} />);
  const button = tree.root.findByType(TouchableOpacity);

  button.props.onPress();

  expect(onPress).toBeCalled();
});

it('should be correct button title', () => {
  const onPress = jest.fn();
  const title = 'Press Me';
  const tree = renderer.create(<Button title={title} onPress={onPress} />);
  const text = tree.root.findByType(Text);

  expect(text.props.children).toBe(title);
});

it('should be primary by default', () => {
  const onPress = jest.fn();
  const title = 'Press Me';
  const tree = renderer
    .create(<Button title={title} onPress={onPress} />)
    .toJSON();

  expect(tree.props.style).toStrictEqual({
    width: '100%',
    height: 45,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#5800ff',
    borderRadius: 11,
    opacity: 1,
  });
});

it('should be correct secondary styles', () => {
  const onPress = jest.fn();
  const title = 'Press Me';
  const tree = renderer
    .create(<Button title={title} onPress={onPress} type="secondary" />)
    .toJSON();

  expect(tree.props.style).toStrictEqual({
    width: '100%',
    height: 45,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
    borderRadius: 11,
    opacity: 1,
    borderColor: '#5800ff',
    borderWidth: 1,
  });
});
