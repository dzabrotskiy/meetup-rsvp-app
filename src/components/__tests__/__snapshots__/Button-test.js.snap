// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#5800ff",
      "borderRadius": 11,
      "display": "flex",
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "fontSize": 17,
          },
          {
            "color": "#fff",
          },
        ],
      ]
    }
  >
    Press Me
  </Text>
</View>
`;
